community/kernel/src/test/java/org/neo4j/kernel/impl/store/MetaDataStoreTest.java

```diff
@@ -542,46 +508,47 @@ public void transactionCommittedMustBeAtomic() throws Exception
                     }
                     while ( cursor.shouldRetry() );
                     assertIdEqualsChecksum( id, checksum, "file" );
+                    fileReadCount.incrementAndGet();
                 }
-            } );
+            } ) ) );
 
-            Runnable apiReader = untilStopped( stopped, runLatch, () -> {
+            race.addContestants( 3, until( end, () ->
+            {
                 TransactionId transaction = store.getLastCommittedTransaction();
                 assertIdEqualsChecksum( transaction.transactionId(), transaction.checksum(), "API" );
-            } );
+                apiReadCount.incrementAndGet();
+            } ) );
 
-            forkMultiple( 10, writer );
-            List<Future<?>> readerFutures = forkMultiple( 5, fileReader );
-            readerFutures.addAll( forkMultiple( 5, apiReader ) );
-
-            runLatch.await( 1, TimeUnit.SECONDS );
-            stopped.set( true );
-
-            for ( Future<?> future : readerFutures )
-            {
-                future.get(); // We assert that this does not throw
-            }
+            race.go();
```

Comment:
same as above here, we need to wait for couple of expected events to happen at least, too aggressive end time can just cancel race